exercises/resistor-colors/test/Tests.hs

```diff
@@ -0,0 +1,44 @@
+{-# OPTIONS_GHC -fno-warn-type-defaults #-}
+{-# LANGUAGE RecordWildCards #-}
+
+import Data.Foldable     (for_)
+import Test.Hspec        (Spec, describe, it, shouldBe)
+import Test.Hspec.Runner (configFastFail, defaultConfig, hspecWith)
+
+import ResistorColors (value)
+
+main :: IO ()
+main = hspecWith defaultConfig {configFastFail = True} specs
+
+specs :: Spec
+specs = describe "value" $ for_ cases test
+  where
+
+    test Case{..} = it explanation assertion
+      where
+        explanation = unwords [show input, "-", description]
+        assertion   = value input `shouldBe` expected
+
+data Case = Case { description :: String
+                 , input       :: [String]
+                 , expected    :: Int
+                 }
+
+cases :: [Case]
+cases = [ Case { description = "Brown and black"
+               , input       = ["brown", "black"]
```

Comment:
Depending on what we want the exercise to teach, we might choose to instead use `Brown` and `Black` here and in all other tests. In that case, we would provide that in the stub.

The similar discussion was had for nucleotide-count in https://github.com/exercism/haskell/issues/694

Reply:
I like this idea!

This exercise was introduced quite recently as a simple exercise that can be put into the core track. If the aim is to put it into the Haskell core track, we can purpose it for anything we like. I don't think we need an extra exercise that teaches monadic error handling (we currently have three already; `collatz-conjecture`, `rna-transcription` and `grains`).

`space-age` already has a data type with multiple constructors, and I use it to teach the difference between pattern matching all constructors vs. dynamic lookup tables, avoiding duplicate code, and naming things. This exercise could do many of the same things without `Float`.

Unless we make it about something that `space-age` isn't, e.g. `Enum`, I am however not sure I see the point in making it a core exercise. @tejasbubane: Let me know what you think wrt. making this a core exercise.

Reply:
Changed.